# GA0004

## Title
Cannot assign an invalid type argument as a type constraint

## Category
Validity

## Severity
Error

## Details
This error is emitted when a type constraint rule contains types that are invalid type arguments. Examples of such are pointer types, the `void` type, etc.

Unbound generic types are excepted from this error.

## Example
```csharp
class C
<
    // GA0004 will appear here
    [PermittedTypes(typeof(int*), typeof(void))]
    T
>
{ }
```
