# GA0005

## Title
Type prohibited from generic type constraints in the constraint clause

## Category
Validity

## Severity
Error

## Details
This error is emitted when a type constraint rule contains types that are invalid type arguments. The erroneous invalid type arguments are pointer types and the `void` type.

Unbound generic types are excepted from this error, in order to support generalization of a type constraint rule for a generic type.

## Example
```csharp
class C
<
    // GA0005 will appear here
    [PermittedTypes(typeof(int*), typeof(void))]
    T
>
{ }
```
